web_src/js/features/codeeditor.ts

```diff
@@ -109,13 +138,20 @@ export async function createMonaco(textarea: HTMLTextAreaElement, filename: stri
     ...other,
   });
 
+  window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', () => {
+    updateTheme(monaco);
+  });
+
   monaco.editor.addKeybindingRules([
     {keybinding: monaco.KeyCode.Enter, command: null}, // disable enter from accepting code completion
   ]);
 
   const model = editor.getModel();
-  model.onDidChangeContent(() => {
-    textarea.value = editor.getValue({preserveBOM: true});
+  model?.onDidChangeContent(() => {
+    textarea.value = editor.getValue({
+      preserveBOM: true,
+      lineEnding: '',
```

Comment:
fyi I verified that the `lineEnding` behaviour should be the same as before:

https://github.com/microsoft/vscode/blob/37f3474c44ac0a2cfc912c0760a24258cc08ebf0/src/vs/editor/browser/widget/codeEditor/codeEditorWidget.ts#L448